spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/BaseBatchReader.java

```diff
@@ -52,6 +55,7 @@
 
   protected CloseableIterable<ColumnarBatch> newBatchIterable(
       InputFile inputFile,
+      ByteBuffer keyMetadata,
```

Comment:
Should this throw an exception if `keyMetadata` is non-null and the format is ORC?